common/src/main/java/com/wynntils/core/consumers/features/FeatureManager.java

```diff
@@ -491,6 +493,23 @@ private void initializeFeature(Feature feature) {
         doEnableFeature(feature);
     }
 
+    private boolean checkStartDisabled(Feature feature) {
+        StartDisabled startDisabledAnnotation = feature.getClass().getAnnotation(StartDisabled.class);
+        if (startDisabledAnnotation == null) return false;
+
+        ConfigProfile[] disabledProfiles = startDisabledAnnotation.disabledProfiles();
+        boolean disableAll = disabledProfiles.length == 0;
+        Set<ConfigProfile> disabledProfileSet = Set.of(disabledProfiles);
+
+        for (ConfigProfile profile : ConfigProfile.values()) {
+            boolean enabledByDefault = !(disableAll || disabledProfileSet.contains(profile));
+            feature.userEnabled.withDefault(profile, enabledByDefault);
```

Comment:
having a hard time understanding this, it seems like we are doing some kind of change to the feature as a side effect here?